benchmark/utils/stencil_matrix.hpp

```diff
@@ -116,19 +116,18 @@ gko::matrix_data<ValueType, IndexType> generate_2d_stencil_box(
                             : discretization_points - i);
     };
 
-    auto flat_idx = [&](const IndexType ix, const IndexType iy) {
+    auto flat_idx = [&](const IndexType iy, const IndexType ix) {
         auto tpx = target_position(ix, positions[0]);
         auto tpy = target_position(iy, positions[1]);
         if (is_in_box(tpx, dims[0]) && is_in_box(tpy, dims[1])) {
-            return global_offset(tpx, tpy) +
-                   target_local_idx(ix) * discretization_points +
-                   target_local_idx(iy);
+            return global_offset(tpy, tpx) + target_local_idx(ix) +
```

Comment:
I reversed the order of the function parameters so that the fastest changing indices are on the right. The position don't matter too much, because it just defines the processor grid. So there is little difference in the end if the grid is [3 x 5] or [5 x 3].